exchanges/kline/kline.go

```diff
@@ -143,6 +145,31 @@ func (i Interval) Short() string {
 	return s
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface for Intervals
+// It does not validate the duration is aligned, only that it is a parsable duration
+func (i *Interval) UnmarshalJSON(text []byte) error {
+	text = bytes.Trim(text, `"`)
+	if len(bytes.TrimLeft(text, `"0123456789`)) > 0 {
+		d, err := time.ParseDuration(string(text))
+		if err != nil {
+			return err
+		}
+		*i = Interval(d)
+	} else {
+		n, err := strconv.ParseInt(string(text), 10, 64)
+		if err != nil {
+			return err
+		}
+		*i = Interval(n)
+	}
+	return nil
```

Comment:
Suggestion: Slightly more performant on both cases, or if not appropriate I think we can rm `"` from the `bytes.TrimLeft(text, `"0123456789`)` check as its removed above. 
```suggestion
	for x := 0; x < len(text); {
		if x == 0 && text[x] == '"' {
			text = text[1 : len(text)-1]
			continue
		}

		if unicode.IsLetter(rune(text[x])) {
			d, err := time.ParseDuration(string(text))
			if err != nil {
				return err
			}
			*i = Interval(d)
			return nil
		}
		x++
	}

	n, err := strconv.ParseInt(string(text), 10, 64)
	if err != nil {
		return err
	}
	*i = Interval(n)
	return nil
```

Reply:
Nice. I like it, but it's denser/more complicated than I'd like.
* Think you may be  missing a cut for the final `"` ?
* Suspect we'd be faster with your approach going x := len(text)-1 backwards, since the letter rune should be at the end
* I'd considered [cutsetting HasSuffix the acceptable unit map](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/time/format.go;l=1573-1581)
I suspect that'd be the fastest. It just meant also inlining a bit more logic than I wanted to care about or own.

Let me play, bench, and come back.
How hot is this, though? I think we might use it enough to make it worth some gentle optimisation.

Reply:
Yeah, Imma leave this alone. Appreciate the swing at it, but it's not hot enough and clarity wins IMO here, preferring to farm to `bytes` over rolling.

Ran some benches:
```
=== RUN   TestUnmarshalJSON
--- PASS: TestUnmarshalJSON (0.00s)
=== RUN   TestUnmarshalJSONV2
--- PASS: TestUnmarshalJSONV2 (0.00s)
=== RUN   TestUnmarshalJSONV3
--- PASS: TestUnmarshalJSONV3 (0.00s)
goos: darwin
goarch: arm64
pkg: github.com/thrasher-corp/gocryptotrader/exchanges/kline
BenchmarkUnmarshalJSON
BenchmarkUnmarshalJSON-8         3171838               376.8 ns/op
BenchmarkUnmarshalJSONV2
BenchmarkUnmarshalJSONV2-8       3554658               341.7 ns/op
BenchmarkUnmarshalJSONV3
BenchmarkUnmarshalJSONV3-8       2849614               401.1 ns/op
PASS
ok      github.com/thrasher-corp/gocryptotrader/exchanges/kline 4.717s
```

Heeded comment about the " in the cutset though.
Fixed 1eeb8570